nixos/modules/services/web-apps/hledger-web.nix

```diff
@@ -26,19 +26,46 @@ in {
       '';
     };
 
-    capabilities = mkOption {
-      type = types.commas;
-      default = "view";
+    capabilities = {
+      view = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Enable the view capability.
+        '';
+      };
+      add = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the add capability.
+        '';
+      };
+      manage = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Enable the manage capability.
+        '';
+      };
+    };
+
+    stateDir = mkOption {
+      type = types.path;
+      default = "/var/lib/hledger-web";
       description = ''
-        Enable the view, add, and/or manage capabilities. E.g. view,add
+        Path the service has access to. If left as the default value this
+        directory will automatically be created before the hledger-web server
+        starts, otherwise the sysadmin is responsible for ensuring the
+        directory exists with appropriate ownership and permissions.
       '';
     };
 
     journalFile = mkOption {
-      type = types.path;
-      example = "/home/hledger/.hledger.journal";
+      type = types.str;
```

Comment:
Reading the documentation of hledger-web again, I've seen that you can define multiple journal files using -f [hledger-web documentation](https://hledger.org/hledger-web.html#files). We can consider making this a listof strings [types.nix](https://github.com/NixOS/nixpkgs/blob/master/lib/types.nix)

Reply:
Sounds reasonable, I added another commit that makes the change. Removed the assertion with it, as users will see an error with their configuration anyway.

@marijanp I you think it's good I can squash the commit into the first one.

Reply:
Looks good too me.